crates/nu-command/src/network/fetch.rs

```diff
@@ -0,0 +1,352 @@
+use base64::encode;
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::IntoPipelineData;
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Value,
+};
+
+use std::path::PathBuf;
+use std::str::FromStr;
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "fetch"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("fetch")
+            .desc("Load from a URL into a cell, convert to table if possible (avoid by appending '--raw').")
+            .required(
+                "URL",
+                SyntaxShape::String,
+                "the URL to fetch the contents from",
+            )
+            .named(
+                "user",
+                SyntaxShape::Any,
+                "the username when authenticating",
+                Some('u'),
+            )
+            .named(
+                "password",
+                SyntaxShape::Any,
+                "the password when authenticating",
+                Some('p'),
+            )
+            .switch("raw", "fetch contents as text rather than a table", Some('r'))
+            .filter()
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Fetch the contents from a URL (HTTP GET operation)."
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
+        run_fetch(engine_state, stack, call, input)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Fetch content from url.com",
+                example: "fetch url.com",
+                result: None,
+            },
+            Example {
+                description: "Fetch content from url.com, with username and password",
+                example: "fetch -u myuser -p mypass url.com",
+                result: None,
+            },
+        ]
+    }
+}
+
+fn run_fetch(
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    _input: PipelineData,
+) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
+    let mut fetch_helper = Fetch::new();
+
+    fetch_helper.setup(engine_state, call, stack)?;
+
+    let runtime = tokio::runtime::Runtime::new()?;
+
+    let path = match fetch_helper.path {
+        Some(p) => p,
+        None => return Err(ShellError::UnsupportedInput("test".to_string(), call.head)),
+    };
+
+    runtime.block_on(fetch(
```

Comment:
I was poking around the other day and saw this: https://github.com/seanmonstar/reqwest#blocking-client

Haven't had a chance to try it, but that might be a way we can use reqwest without also pulling in tokio and async/await into e-q. There's also https://docs.rs/reqwest/latest/reqwest/blocking/struct.Response.html#method.copy_to which might let us do the streaming. I noticed that `.bytes()` seems to grab the whole response, which for large downloads might take a while before we can put anything into the pipeline.

Reply:
We could do that, for sure. If we're not set on using `reqwest`, we could try `ureq` crate instead. It's a simpler and a bit lighter library than `reqwest`, though it does not support async and thus it does not require tokio. 

You would suggest to use `copy_to` for anything that requires to get the bytes back? Would that mean that we'd not convert the body of the request into a `value`, but instead use a stream and get the bytes from the request?